File: docflow/backend/src/main/java/com/docflow/reports/service/ReportRunPolicy.java
package com.docflow.reports.service;

import org.springframework.http.HttpStatus;
import org.springframework.web.server.ResponseStatusException;

import java.util.List;

public final class ReportRunPolicy {

    public static final boolean REQUIRE_AT_LEAST_ONE_RUNTIME_FILTER = true;
    public static final String AT_LEAST_ONE_FILTER_MESSAGE = "At least one filter value is required to run the report.";

    private ReportRunPolicy() {
    }

    public static void validateAtLeastOneRuntimeFilter(List<?> filters) {
        if (!REQUIRE_AT_LEAST_ONE_RUNTIME_FILTER) {
            return;
        }
        if (filters == null || filters.isEmpty()) {
            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, AT_LEAST_ONE_FILTER_MESSAGE);
        }
    }
}

File: docflow/frontend/src/lib/reportRunPolicy.ts
export const REQUIRE_AT_LEAST_ONE_REPORT_FILTER = true;
export const AT_LEAST_ONE_REPORT_FILTER_MESSAGE = 'At least one filter value is required to run the report.';

export function validateAtLeastOneReportFilter(filters: unknown[] | null | undefined): string | null {
  if (!REQUIRE_AT_LEAST_ONE_REPORT_FILTER) {
    return null;
  }
  return filters && filters.length > 0 ? null : AT_LEAST_ONE_REPORT_FILTER_MESSAGE;
}
